packages/core/src/modules/proofs/protocol/v1/__tests__/indy-proof-presentation.test.ts

```diff
@@ -0,0 +1,252 @@
+import type { Agent } from '../../../../../agent/Agent'
+import type { ConnectionRecord } from '../../../../connections/repository/ConnectionRecord'
+import type {
+  AcceptPresentationOptions,
+  AcceptProposalOptions,
+  ProposeProofOptions,
+} from '../../../models/ModuleOptions'
+import type { ProofRecord } from '../../../repository/ProofRecord'
+import type { PresentationPreview } from '../models/V1PresentationPreview'
+
+import { setupProofsTest, waitForProofRecord } from '../../../../../../tests/helpers'
+import testLogger from '../../../../../../tests/logger'
+import { DidCommMessageRepository } from '../../../../../storage/didcomm'
+import { ProofProtocolVersion } from '../../../models/ProofProtocolVersion'
+import { ProofState } from '../../../models/ProofState'
+import { V1PresentationMessage, V1ProposePresentationMessage, V1RequestPresentationMessage } from '../messages'
+
+describe('Present Proof', () => {
+  let faberAgent: Agent
+  let aliceAgent: Agent
+  let aliceConnection: ConnectionRecord
+  let presentationPreview: PresentationPreview
+  let faberProofRecord: ProofRecord
+  let aliceProofRecord: ProofRecord
+  let didCommMessageRepository: DidCommMessageRepository
+
+  beforeAll(async () => {
+    testLogger.test('Initializing the agents')
+    ;({ faberAgent, aliceAgent, aliceConnection, presentationPreview } = await setupProofsTest(
+      'Faber agent',
+      'Alice agent'
+    ))
+  })
+
+  afterAll(async () => {
+    testLogger.test('Shutting down both agents')
+    await faberAgent.shutdown()
+    await faberAgent.wallet.delete()
+    await aliceAgent.shutdown()
+    await aliceAgent.wallet.delete()
+  })
+
+  test(`Alice Creates and sends Proof Proposal to Faber`, async () => {
+    testLogger.test('Alice sends proof proposal to Faber')
+
+    const proposeOptions: ProposeProofOptions = {
+      connectionId: aliceConnection.id,
+      protocolVersion: ProofProtocolVersion.V1,
+      proofFormats: {
+        indy: {
+          name: 'ProofRequest',
+          nonce: '58d223e5-fc4d-4448-b74c-5eb11c6b558f',
+          version: '1.0',
+          attributes: presentationPreview.attributes,
+          predicates: presentationPreview.predicates,
+        },
+      },
+      comment: 'V1 propose proof test',
+    }
+
+    aliceProofRecord = await aliceAgent.proofs.proposeProof(proposeOptions)
+
+    testLogger.test('Faber waits for presentation from Alice')
+    faberProofRecord = await waitForProofRecord(faberAgent, {
+      threadId: aliceProofRecord.threadId,
+      state: ProofState.ProposalReceived,
+    })
```

Comment:
For initial method (proposal / request) we will not be having ProofRecord so it will be throwing error at line 63 as aliceProofRecord is receiving value at line 67